### ISSUE TITLE
Numeric operators with a single element throws type error on non number types

### ISSUE THREAD

**Issue by andrewchambers** [janet-lang/janet#270]
```
 (+ (int/s64 1))
error: expected number, got core/s64
  in _thunk [repl] (tailcall) on line 2, column 1
```

**Comment by bakpakin** [comment]
This is because unary numeric operations are transformed at compile to:

`(+ 0 (int/s64 1))`

There are some ways around this, and this should definitely be better. For example, `(- (int/s64 1))` won't work well either, and neither will `(/ (int/s64))`.

**Comment by bakpakin** [comment]
Added overloading for right-handed operands as well, a la python.

For example, types can define :+ as well as :r+ methods.
Fixed in 4fe005e3c3ea0d86ef583bb756e02c7572910e4d
